*: bump github.com/knz/go-libedit

petermattis opened the pull request:
This picks up a change which silences the pointer-sign conversion warnings which were in turn causing `TestLint/TestVet` to error spuriously.

Fixes #34059

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/34069)
<!-- Reviewable:end -->

petermattis commented:
Ugh, I can't whack vendor like this without the upstream change (we check that the contents of vendor match `Gopkg.lock`). I can't merge the upstream change without @knz's approval: https://github.com/knz/go-libedit/pull/15.

petermattis committed:
*: bump github.com/knz/go-libedit

This picks up a change which silences the pointer-sign conversion
warnings which were in turn causing `TestLint/TestVet` to error
spuriously.

Fixes #34059

Release note: None

petermattis commented:
I updated this PR to bump the version of `github.com/knz/go-libedit` we're using, and to unskip `TestLint/TestVet`. PTAL.

petermattis commented:
bors r=jordanlewis,knz
